app/src/main/java/com/pvp/app/ui/screen/filter/Filter.kt

```diff
@@ -0,0 +1,132 @@
+    val user = model.user.collectAsStateWithLifecycle()
+
+    LaunchedEffect(user.value) {
+        selectedActivities = user.value?.activities ?: emptyList()
+        unselectedActivities = activities - selectedActivities
+    }
+
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(16.dp)
+            .verticalScroll(rememberScrollState())
+    ) {
+        ActivitiesBox(
+            title = "Selected activities",
+            activities = selectedActivities,
+            onClick = { activity ->
+                selectedActivities = selectedActivities.minus(activity)
+                unselectedActivities = unselectedActivities.plus(activity)
+            }
+        )
+
+        ActivitiesBox(
+            title = "Available activities",
+            activities = unselectedActivities,
+            onClick = { activity ->
+                unselectedActivities = unselectedActivities.minus(activity)
+                selectedActivities = selectedActivities.plus(activity)
+            }
+        )
+
+        Button(
+            onClick = {
+                model.updateUserActivities(selectedActivities)
+                /* TODO: redirect back to profile? page */
+            },
+            modifier = Modifier.align(Alignment.CenterHorizontally)
+        ) {
+            Text(
+                text = "Update",
+                style = MaterialTheme.typography.bodyMedium
+            )
+        }
+    }
+}
+
+@OptIn(ExperimentalLayoutApi::class)
+@Composable
+fun ActivitiesBox(
+    title: String,
+    activities: List<String>,
+    onClick: (String) -> Unit
+) {
+    Box(
+        modifier = Modifier
+            .fillMaxWidth()
+            .background(MaterialTheme.colorScheme.surface)
+            .padding(8.dp)
+    ) {
+        Column(
+            modifier = Modifier.fillMaxWidth()
+        ) {
+            Text(
+                text = title,
+                style = TextStyle(fontSize = 18.sp),
+                modifier = Modifier.padding(8.dp)
+            )
+            FlowRow(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(8.dp)
+            ) {
+                activities
+                    .sorted()
+                    .forEach { activity ->
+                    Card(
+                        modifier = Modifier
+                            .padding(8.dp)
+                            .background(
+                                MaterialTheme.colorScheme.secondary,
+                                MaterialTheme.shapes.medium
+                            ),
+                        onClick = { onClick(activity) }
+                    ) {
+                        Text(
+                            text = activity,
+                            style = TextStyle(fontSize = 16.sp),
+                            modifier = Modifier.padding(16.dp)
+                        )
+                    }
```

Comment:
Whole card with its children components may not be correctly indented. `Control+Alt+L` would fix this. I think 4 spaces are missing.